test/application_view_test_case.rb

```diff
@@ -10,6 +10,21 @@ def assert_attrs(tag, tag_name: :input, **attrs)
     assert_match /<#{tag_name}.* #{attrs}/, tag
   end
 
+  def mock_form(form_object: OpenStruct.new)
```

Comment:
I understand the instinct to stub this out in the spirit of encapsulation.

In this case I'd rather not stub the form builder methods since they don't "belong" to us.

Let's test the real thing instead. SQLite supports JSON columns. Let's add one to one of the fixtures instead and use that in the tests instead of mocking the form.